apis/sql/v1alpha2/zz_server_types.go

```diff
@@ -31,19 +31,19 @@ type ServerExtendedAuditingPolicyObservation struct {
 type ServerExtendedAuditingPolicyParameters struct {
 
 	// +kubebuilder:validation:Optional
-	LogMonitoringEnabled *bool `json:"logMonitoringEnabled,omitempty" tf:"log_monitoring_enabled,omitempty"`
+	LogMonitoringEnabled *bool `json:"logMonitoringEnabled,omitempty" tf:"log_monitoring_enabled"`
 
 	// +kubebuilder:validation:Optional
-	RetentionInDays *int64 `json:"retentionInDays,omitempty" tf:"retention_in_days,omitempty"`
+	RetentionInDays *float64 `json:"retentionInDays,omitempty" tf:"retention_in_days"`
 
 	// +kubebuilder:validation:Optional
-	StorageAccountAccessKeyIsSecondary *bool `json:"storageAccountAccessKeyIsSecondary,omitempty" tf:"storage_account_access_key_is_secondary,omitempty"`
+	StorageAccountAccessKey *string `json:"storageAccountAccessKey,omitempty" tf:"storage_account_access_key"`
```

Comment:
This looks not expected. Guessing it is not properly marked as sensitive in cli schema, could you please check that?

I also saw another one but cannot find now. It would be great if you could check other types.

Reply:
after checking again, I noticed several differences with `GetConnectionDetailsMapping` functions. @turkenh is this expected?

```
 func (tr *Server) GetConnectionDetailsMapping() map[string]string {
-       return map[string]string{"administrator_login_password": "spec.forProvider.administratorLoginPasswordSecretRef", "extended_auditing_policy[*].storage_account_access_key": "spec.forProvider.extendedAuditingPolicy[*].storageAccountAccessKeySecretRef", "threat_detection_policy[*].storage_account_access_key": "spec.forProvider.threatDetectionPolicy[*].storageAccountAccessKeySecretRef"}
+       return map[string]string{"administrator_login_password": "spec.forProvider.administratorLoginPasswordSecretRef", "threat_detection_policy[*].storage_account_access_key": "spec.forProvider.threatDetectionPolicy[*].storageAccountAccessKeySecretRef"}
 }

```

Reply:
@turkenh you're right. It's not marked as sensitive
```
              "storage_account_access_key": {
                "type": "string",
                "description_kind": "plain",
                "optional": true,
                "sensitive": true
              },
              "storage_account_access_key_is_secondary": {
                "type": "bool",
                "description_kind": "plain",
                "optional": true
              },
```

Reply:
@ezgidemirel the diff here is a bit misleading btw. 

There are two changes here:

1. For `StorageAccountAccessKeyIsSecondary` which is a `bool`, `omitempty` is dropped from `tf` tag. **This is expected** and not because of cli shema change but due to this PR: https://github.com/crossplane/terrajet/pull/244/files
2.`StorageAccountAccessKeySecretRef` is changed as `StorageAccountAccessKey` since it is not marked as **sensitive**. I wanted to make sure that whether this is some information that we could get/infer using the schema.json but looks like it is not. Here is how `azurerm_mssql_server.extended_auditing_policy` looks like:

```
                            "extended_auditing_policy": {
                                "type": [
                                    "list",
                                    [
                                        "object",
                                        {
                                            "log_monitoring_enabled": "bool",
                                            "retention_in_days": "number",
                                            "storage_account_access_key": "string",
                                            "storage_account_access_key_is_secondary": "bool",
                                            "storage_endpoint": "string"
                                        }
                                    ]
                                ],
                                "description_kind": "plain",
                                "deprecated": true,
                                "optional": true,
                                "computed": true
                            },
```
As you can see, there is no more information for `storage_account_access_key` other than being a `string`. Please note the schema you've shared above is probably from another resource and I don't think we would want to complicate the code generation pipeline to infer information using the schema of another resource. Hence I would propose to override the schema in such cases with a custom config as documented here: https://github.com/crossplane/terrajet/blob/main/docs/configuring-a-resource.md#overriding-terraform-resource-schema. For deeply nested objects you can check this as an example: https://github.com/crossplane-contrib/provider-jet-gcp/pull/2/files#diff-945780a6043d1fd432785f94ae1b60f755d620640454cb50f57f9a292f78e857R24-R26